src/json2daisy/json2daisy.py

```diff
@@ -178,7 +178,7 @@ def generate_header(board_description_dict: dict) -> 'tuple[str, dict]':
         raise NameError(f'Unkown som "{som}"')
 
     # alphabetize by component name
-    components = sorted(components.items(), key=lambda x: x[1]['component'])
+    components = sorted(components.items(), key=lambda x: x[1]['component'] + x[0])
```

Comment:
minor bug fix with the object array generation. I didn't realize this is only sorting by component type, not the name of the component. This change makes it so that knob1 will always be before knob2 for example